TourmalineCore/pelican-ui
# feat: support playwright in Dev Container with proper documentation and ci pipeline speeding up

These changes add Playwright support to our Dev Container using VNC, dedicated docker image with included browsers and Playwright deps (but no playwright itself!)

**Used Refs**
- https://github.com/cypress-io/cypress-documentation/issues/2956#issuecomment-1691335689
- https://github.com/TourmalineCore/to-dos-ui/commit/2d59b6415f1de3adb000c637bbf4d7c34f7d8c08#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R47
- https://www.oddbird.net/2022/11/30/headed-playwright-in-docker/
- https://playwright.dev/docs/docker
- https://mcr.microsoft.com/en-us/artifact/mar/playwright/tags

**How to Verify**
You need to open Terminal from inside Dev Container.

- run `npm ci` to install all deps with the fixed versions of playwright packages
- run `npm run test` to start the project  with static content for tests (don't stop and and keep the terminal alive)
- in a new Terminal instance run `npx playwright test` to see if tests run in headless mode (no browser is opening)
- open VNC of Dev container per [these](https://github.com/TourmalineCore/to-dos-ui#cypress-dev-containers-only) instructions
-  in a new Terminal instance run `npx playwright test --ui` => you should see opening Playwright in VNC and be able to run the tests
